sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -41,6 +41,8 @@ trait Catalog {
 
   def unregisterAllTables(): Unit
 
+  def showTables(databaseName: Option[String]): immutable.List[String]
```

Comment:
I think we should probably stick to generic `Seq` in public interfaces like this.